gitpod-io/gitpod
# Implement `ListOrganizationPrebuilds`

## Description

Implements the `ListOrganizationPrebuilds` method for `PrebuildService`. This allows us to lay the groundwork for global prebuilds history and migrate the old `ListPrebuilds` usages to this new method.

We will not be removing the `ListPrebuilds` method as anticipated, because it is being used in `PrebuildLogs.tsx` with an incompatible `workspaceId` parameter, which we will be able to hopefully replace by the time we introduce the updated `GetPrebuild` method.

## Related Issue(s)

Fixes EXP-1227

## How to test

1. You can [join my org](https://ft-impleemfb9de79866.preview.gitpod-dev.com/orgs/join?inviteId=12b7907c-b706-447a-ab36-dbb7b9e14ac7) to get some prebuilds out-of-the-box

You can try this as an example for filtering based on a search, but make sure that you provide an auth header or cookie with a token. [^1]

```bash
curl 'https://ft-impleemfb9de79866.preview.gitpod-dev.com/public-api/gitpod.v1.PrebuildService/ListOrganizationPrebuilds' -X POST -H 'content-type: application/json' --data-raw '{"organizationId":"7075b766-b913-439c-853e-a6270cf7c9d7", "filter": {"searchTerm": "gitpod"}}' | jq '.prebuilds | length'
```

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ft-impleemfb9de79866</li>
	<li><b>🔗 URL</b> - <a href="https://ft-impleemfb9de79866.preview.gitpod-dev.com/workspaces" target="_blank">ft-impleemfb9de79866.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - ft-impleement-global-prebuild-list-gha.22373</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-ft-impleemfb9de79866%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

[^1]: Personally, I've found it the easiest to get a proper curl request with all of the auth headers by inspecting a request the dashboard makes for any call, right-clicking and doing `Copy as cURL` (or similar)
